### DISCUSSION TITLE
No "ModBR Tx" and "ModBR Rx" options for Modbus Configuration.

### DISCUSSION THREAD

**Discussion by mcavalcante2** [arendst/Tasmota#19966] [Q&A]
I´m folling the steps of he https://tasmota.github.io/docs/Modbus-Bridge/ after installing the binary with Modbus enable, however, the options "ModBR Tx" and "ModBR Rx" don´t appears for me in the combo box as suggested by the procedure.

To build the binary, I´ve used TasmoCompiler v11.1.0 and, in "Select features", I´ve selected Modbus Bridge. My device is a ESP32-WROOM-32U.

Does anyone have experienced this? How to setup Modbus with Tasmota?

**Comment by mcavalcante2** [comment]
Found the path. When compiling the firmware thru Gitpod, besides enabling the Modbus bridge in the check boxes, it´s also need to enable these parameters:
			#ifndef USE_MODBUS_BRIDGE
			#define USE_MODBUS_BRIDGE       // Add support for software Modbus Bridge (+3k code)
			#define USE_MODBUS_BRIDGE_TCP   // Add support for software Modbus TCP Bridge (Must also enable USE_MODBUS_BRIDGE)
			#endif

Anyway, could not bring the TCP bridge up, even with an RTU golden device connected to the RS485 network.

**Reply by RaW79** [reply]
Have you assigned the Modbus Rx Tx GPIOs (Modbus Bridge Rx Tx) also when using TCP in the device configuration settings?

Evaluating at the moment by myself how the modbus bridge is to be used and after assignmend of the Rx Tx I could startup Modbus TCP. Used a compiled version from tasmocompiler including your parameters.
